src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/SharedUtils.java

```diff
@@ -415,13 +414,12 @@ static MethodHandle wrapWithAllocator(MethodHandle specializedHandle,
     // lazy init MH_ALLOC and MH_FREE handles
     private static class AllocHolder {
 
-        static final LibraryLookup LOOKUP = LibraryLookup.ofDefault();
-
-        static final MethodHandle MH_MALLOC = getSystemLinker().downcallHandle(LOOKUP.lookup("malloc").get(),
+        private static final CLinker linker = getSystemLinker();
+        static final MethodHandle MH_MALLOC = linker.downcallHandle(CLinker.findNative("malloc").get(),
                         MethodType.methodType(MemoryAddress.class, long.class),
                 FunctionDescriptor.of(C_POINTER, C_LONG_LONG));
 
-        static final MethodHandle MH_FREE = getSystemLinker().downcallHandle(LOOKUP.lookup("free").get(),
+        static final MethodHandle MH_FREE = linker.downcallHandle(CLinker.findNative("free").get(),
```

Comment:
These symbol lookups fail on Windows.

Reply:
Following fix works:

```
diff --git a/src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/SharedUtils.java b/src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/SharedUtils.java
index d705d414c0d..708c005d1d9 100644
--- a/src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/SharedUtils.java
+++ b/src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/SharedUtils.java
@@ -50,6 +50,7 @@ import java.lang.invoke.MethodType;
 import java.lang.invoke.VarHandle;
 import java.lang.ref.Reference;
 import java.nio.charset.Charset;
+import java.nio.file.Path;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Consumer;
@@ -415,6 +416,9 @@ public class SharedUtils {
     private static class AllocHolder {

         private static final CLinker linker = getSystemLinker();
+        static {
+            System.load(Path.of(System.getenv("SystemRoot"), "System32", "msvcrt.dll").toAbsolutePath().toString());
+        }
         static final MethodHandle MH_MALLOC = linker.downcallHandle(CLinker.findNative("malloc").get(),
                         MethodType.methodType(MemoryAddress.class, long.class),
                 FunctionDescriptor.of(C_POINTER, C_LONG_LONG));
```

Reply:
Yes, will add this as temporary hack for Windows.